### PR TITLE
[Bug] Diamond Storm should only trigger once when hitting multiple pokemon

### PR DESCRIPTION
**Pull request by DayKev**
## What are the changes the user will see?
Diamond Storm will only give the defense buff once per use, even if it hits 2 Pokémon.

## Why am I making these changes?
It currently applies the defense buff per pokemon hit instead of 

## What are the changes from a developer perspective?
The `firstTargetOnly` param is set to `true` for Diamond Storm's `StatStageChangeAttr`.

## How to test the changes?
`npm run test diamond_storm`

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I considered writing automated tests for the issue?
- ~[ ] If I have text, did I make it translatable and add a key in the English locale file(s)?~
- ~[ ] Have I tested the changes (manually)?~
    - [x] Are all unit tests still passing? (`npm run test`)
- ~[ ] Are the changes visual?~
  - ~[ ] Have I provided screenshots/videos of the changes?~

### CODE REVIEW

**Review by frutescens — DISMISSED** [review]

**Comment by Snailman11** [comment]
https://m.bulbapedia.bulbagarden.net/wiki/Diamond_Storm_(move)
Are you sure that removing per target stat increases is accurate to cart? Have you tested current implementation?

**Comment by Snailman11** [comment]
Link to Discord Post (Spread Moves that Raise Stats Trigger Twice Upon Hitting Two Pokemon)
https://discord.com/channels/1125469663833370665/1290907835797995530

**Review by innerthunder — DISMISSED** [review]

**Review by innerthunder — APPROVED** [review]

File: src/test/moves/diamond_storm.test.ts
Comment (torranx) [inline src/test/moves/diamond_storm.test.ts:1]: would love to see all tests for the [move based on the description](https://bulbapedia.bulbagarden.net/wiki/Diamond_Storm_(move))

Comment (DayKev) [reply src/test/moves/diamond_storm.test.ts:1]: There's no extra tests to add, the stuff mentioned at the top part of the description is from the old behavior before it got changed (and PokeRogue uses the latest behavior).

File: src/test/moves/diamond_storm.test.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      .enemyAbility(Abilities.BALL_FETCH)
+      .enemyMoveset(Moves.SPLASH);
+  });
+
+  it("should only increase defense once even if hitting 2 pokemon", async () => {
+    game.override.battleType("double");
+    const diamondStorm = allMoves[Moves.DIAMOND_STORM];
+    vi.spyOn(diamondStorm, "chance", "get").mockReturnValue(100);
+    vi.spyOn(diamondStorm, "accuracy", "get").mockReturnValue(100);
+    await game.classicMode.startBattle([Species.FEEBAS]);
```
Comment (DayKev) [inline src/test/moves/diamond_storm.test.ts:39]:
```suggestion
    await game.classicMode.startBattle([ Species.FEEBAS ]);
```

File: src/test/moves/diamond_storm.test.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  });
+
+  afterEach(() => {
+    game.phaseInterceptor.restoreOg();
+  });
+
+  beforeEach(() => {
+    game = new GameManager(phaserGame);
+    game.override
+      .moveset([Moves.DIAMOND_STORM])
```
Comment (DayKev) [inline src/test/moves/diamond_storm.test.ts:27]:
```suggestion
      .moveset([ Moves.DIAMOND_STORM ])
```

**Review by innerthunder — APPROVED** [review]

**Review by torranx — APPROVED** [review]

**Pull request merged**